src/NumberTheory/vinberg.jl

```diff
@@ -0,0 +1,322 @@
+###################################################################
+# Vinberg's algorithm
+###################################################################
+
+@doc raw"""
+    _check_v0(Q::ZZMatrix, v0::ZZMatrix) -> ZZMatrix
+
+Checks if the inner product of the given `v0` is positive or generates such a `v0` if none is given.
+"""
+function _check_v0(Q::ZZMatrix, v0::ZZMatrix)
+  if iszero(v0)
+    Eigenvalues = eigenvalues(QQ, Q)
+    stopping_condition = 0
+    negative_eigenvalue = 0
+    for x in Eigenvalues
+      if x < 0
+        negative_eigenvalue = x
+        break
+      end
+      stopping_condition =+
```

Comment:
Looks as if something is wrong here? Perhaps this was meant?
```suggestion
      stopping_condition += 1
```